modular_skyrat/modules/modular_implants/code/soulcatcher/soulcatcher_component.dm

```diff
@@ -339,6 +345,52 @@ GLOBAL_LIST_EMPTY(soulcatchers)
 	room_to_join.add_soul_from_ghost(src)
 	return TRUE
 
+/**
+ * Args:
+ * datum/component/soulcatcher/catcher - The soulcatcher we are checking.
+ *
+ * Returns: TRUE by default.
+ */
+/mob/proc/can_join_soulcatcher(datum/component/soulcatcher/catcher)
+	return TRUE
+
+/**
+ * Args:
+ * datum/component/soulcatcher/catcher - The soulcatcher we are checking.
+ *
+ * Returns: False if super returned false, false if the catcher is not joinable by ghosts, true otherwise.
+ */
+/mob/dead/observer/can_join_soulcatcher(datum/component/soulcatcher/catcher)
+	. = ..()
+
+	if (!.)
+		return
+
+	if (!catcher.ghost_joinable)
+		return FALSE
+
+/**
+ * Args:
+ * datum/soulcatcher_room/room - The room we are checking.
+ * check_parent = TRUE - If TRUE, returns false if no catcher is room's parent, as well as if can_join_soulcatcher(catcher) returns false.
+ * datum/component/soulcatcher/catcher - Only present for efficiency - if running this proc in a catcher's loop, put src in here.
+ *
+ * Returns: False if check_parent is TRUE and no catcher is present or can_join_soulcatcher(catcher) returns false, false if the room is unjoinable, true otherwise.
+ */
+/mob/proc/can_join_soulcatcher_room(datum/soulcatcher_room/room, check_parent = TRUE, datum/component/soulcatcher/catcher)
+	if (check_parent)
+		if (!catcher)
+			catcher = room.master_soulcatcher.resolve()
+			if (!catcher)
+				return FALSE
+		if (!can_join_soulcatcher(catcher))
+			return FALSE
+
+	if (!room.joinable)
+		return FALSE
```

Comment:
```suggestion
	if(!room.joinable)
		return FALSE
	else if(!check_parent)
	        return TRUE 
	
	if(!catcher)
		catcher = room.master_soulcatcher.resolve()
		if(!catcher)
			return FALSE
	
	if(!can_join_soulcatcher(catcher))
			return FALSE

```
we want to avoid nesting.

Reply:
fair, this is the more efficient line of logic anyway